pkg/e2e/operators/ocmagent/ocmagent.go

```diff
@@ -0,0 +1,235 @@
+package operators
+
+import (
+	"context"
+	"strings"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/osde2e/pkg/common/alert"
+	"github.com/openshift/osde2e/pkg/common/expect"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	"github.com/openshift/osde2e/pkg/common/label"
+	"github.com/openshift/osde2e/pkg/e2e/operators"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	operatorhubv1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+)
+
+var suiteName = "OCM Agent Operator"
+
+func init() {
+	alert.RegisterGinkgoAlert(suiteName, "SD_SREP", "@ocm-agent-operator", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+// TODO: Separate PR to remove 'Informing' label from test suite
+var _ = ginkgo.Describe(suiteName, ginkgo.Ordered, label.Operators, label.Informing, func() {
+	var h *helper.H
+	var client *resources.Resources
+
+	ginkgo.BeforeAll(func() {
+		h = helper.New()
+		client = h.AsUser("")
+	})
+
+	var (
+		configMapName      = "ocm-agent-config"
+		clusterRolePrefix  = "ocm-agent-operator"
+		deploymentName     = "ocm-agent"
+		namespace          = "openshift-ocm-agent-operator"
+		networkPolicyName  = "ocm-agent-allow-only-alertmanager"
+		operatorName       = "ocm-agent-operator"
+		operatorRegistry   = "ocm-agent-operator-registry"
+		secretName         = "ocm-access-token"
+		serviceMonitorName = "ocm-agent-metrics"
+		serviceName        = "ocm-agent"
+
+		deployments = []string{
+			deploymentName,
+			deploymentName + "-operator",
+		}
+	)
+
+	ginkgo.It("cluster service version exists", label.Install, func(ctx context.Context) {
+		var csvs operatorhubv1.ClusterServiceVersionList
+		err := client.List(ctx, &csvs, resources.WithFieldSelector(labels.FormatLabels(map[string]string{"metadata.namespace": namespace})))
```

Comment:
Can this be just a get with a field selector of `spec.displayname`?

Reply:
@jbpratt `Get` method requires the name of the resource (which we do not know as the resource name contains additional data beyond the standard string for the csv `ocm-agent-operator-...`.

When using `List` function to attempt to filter on other labels or fields (e.g. `spec.displayName`), it is not supported.

```
              Status: "Failure",
              Message: "field label not supported: spec.displayName",
              Reason: "BadRequest",
              Details: nil,
              Code: 400,
```

Unless I am doing something incorrect, are we okay to leave this as is? Either way, we would need to use the `List` function which returns a list and we would either need to loop over it or get the specific position in the list (assuming we would always get one item back).